substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/concurrent/DeferredCommonPool.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2017, 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.concurrent;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Pure delegate implementation to ForkJoinPool.commonPool().
+ */
+public class DeferredCommonPool extends ForkJoinPool {
+
+    public DeferredCommonPool() {
+        super(1);
```

Comment:
The delegate mechanism is an interesting idea. My concern is that we miss some necessary overrides: what if a new version of the JDK adds additional methods to `ForkJoinPool`? We need some safeguards. My first idea was to also override some internal methods of `ForkJoinPool`, i.e., the methods that actually start a thread / modify the queue, and throw an exception in those overrides. But they all seem to be private or package private. So the best I can come up with is to use the super constructor here that also provides a thread factory, and pass in a custom thread factory that throws an exception when a new thread is requested. That way we at least have the guarantee that the delegate pool does not start a thread itself.

Reply:
I'll look into that.